keps/sig-cli/0031-kustomize-integration.md

```diff
@@ -32,122 +33,210 @@ superseded-by:
+- SIG CLI have expertise in the Kustomize codebase and is committed to maintaining this solution going forward.
+- Providing it as part of kubectl will ensure that it is available to users of kubectl apply.
 
 ## Proposal
 
-### UX
+Do one or more of these.
+
+- [ ] A: `kubectl kustomize -f dir/ | kubectl apply -f -`
+- [ ] B: `kubectl apply -f dir/`
+- [ ] C: `kubectl apply -f dir/kustomization.yaml` (similar to B, but slightly different UX)
+
+### Option A: Kustomize Command
+
+Publish the `kustomize build` command as `kubectl kustomize`.  Update documentation demonstrate using Kustomize as
+`kubectl kustomize -f dir/ | kubectl kustomize -f -`.  Consider Option B as a follow up if there is consensus on
+the reason for doing so and trade-offs.
+
+#### Why we like this approach
+
+Kustomize does more than the resource builder in kubectl (e.g. the thing that processes files provided with `-f`).
+
+- It is consistent with how tools that produce Resource Config but exist outside kubectl would integrate with
+ `kubectl apply`
+- It is clear to the user that they are getting a new behavior than when they ran `kubectl apply -f dir/`
+ which was relatively restrictive in terms of what it did.
+- Publishing it as a separate command keeps what the other kubectl commands do simpler
+- We have talked about moving towards this approach for how we teach users to work with imperative generators -
+ e.g. `kubectl create deployment -o yaml --dry-run | kubectl apply -f -` (or something that doesn't exist yet like
+  - **Note:** this case still has friction (e.g. it defaults fields - such as creationTimestamp to null - yuck)
+ `kubectl generate deployment | kubectl apply -f -`)
+- It makes more sense to first provide a command (Option A) and then integrate into resource builder (Option B)
+  than it does to do the reverse.
 
-When apply, get or delete is run on a directory, check if it contains a kustomization.yaml file. If there is, apply, get or delete the output of kustomize build. Kubectl behaves the same as current for directories without kustomization.yaml.
+#### Why don't like this approach
 
-#### apply
+- It can't address user friction requiring deeper integration - e.g. produce meaningful line numbers in error
+  messages.
+- Most commands would require the same input pipe - e.g. get, delete, etc would all need pipes
+- We haven't trained user with this pattern
+ - We don't currently follow this pattern by default for other commands such as `kubectl create deployment`- which requires
+   additional flags to output content to pipe `--dry-run -o yaml`
+ - We don't do this for file, dir or url targets - e.g. we don't do `curl something | kubectl apply -f -`
+- When demoed, the UX was considered more complicated than directly integrating into the resource builder
+- It’s more typing for the user
+- Kustomize is the way we are addressing issues with declarative workflows so it should be integrated into them.
```

Comment:
I have wondered when to capitalize kustomize. Here it's at the beginning of the sentence, but elsewhere it's still written as all lowercase even in this position. That's more of a branding concern, but this caught my eye.